src/super_gradients/common/sg_loggers/dagshub_sg_logger.py

```diff
@@ -166,28 +167,69 @@ def _dvc_add(self, local_path="", remote_path=""):
     def _dvc_commit(self, commit=""):
         self.dvc_folder.commit(commit, versioning="dvc", force=True)
 
+    @multi_process_safe
+    def _get_nested_dict_values(self, d, parent_key="", sep="/"):
+        items = []
+        for k, v in d.items():
+            new_key = parent_key + sep + k if parent_key else k
+            if isinstance(v, Mapping):
+                items.extend(self._get_nested_dict_values(v, new_key, sep=sep))
+            else:
+                items.append((new_key, v))
+        return items
+
+    @multi_process_safe
+    def _contains_special_characters(self, text):
+        pattern = r"[!\"#$%&'()*+,:;<=>?@[\]^`{|}~\t\n\r\x0b\x0c]"
+        matches = re.findall(pattern, text)
+        if matches:
+            return True, ", ".join(matches)
+        return False, None
+
     @multi_process_safe
     def add_config(self, tag: str, config: dict):
         super(DagsHubSGLogger, self).add_config(tag=tag, config=config)
-        param_keys = config.keys()
-        for pk in param_keys:
-            for k, v in config[pk].items():
-                try:
-                    mlflow.log_params({k: v})
-                except Exception:
-                    logger.warning(f"Skip to log {k}: {v}")
+        flatten_dict = self._get_nested_dict_values(d=config)
+        for k, v in flatten_dict:
+            try:
+                mlflow.log_params({k: v})
+            except Exception as e:
+                is_contain, spec_char = self._contains_special_characters(k)
+                if is_contain:
+                    err_msg = f"Fail to log {k}, please remove the unsupported characters: {spec_char}"
```

Comment:
Hmm, I doubt that is the right thing to do (asking user to fix some parameter, especially if it is coming from SG codebase). 
Maybe it's better to just sanitize the key and remove/replace unsupported characters with "_" or whatever and still log it?
Another option is to support all characters in MLFlow ;) 
WDYT?